src/commands/email/send.js

```diff
@@ -8,27 +8,86 @@ const path = require('path');
 class Send extends BaseCommand {
   async run() {
     await super.run();
+
     if (!process.env.SENDGRID_API_KEY) {
-      this.logger.error('Make sure you have an environment variable called SENDGRID_API_KEY set up with your SendGrid API key. Visit https://app.sendgrid.com/settings/api_keys to get an API key.');
+      this.logger.error(
+        'Make sure you have an environment variable called SENDGRID_API_KEY set up with your SendGrid API key. Visit https://app.sendgrid.com/settings/api_keys to get an API key.'
+      );
       return this.exit(1);
     }
+
+    const pipedInput = await this.readStream();
+    if (pipedInput) {
+      this.processData(pipedInput);
+    }
+
+    this.isTTY = process.stdin.isTTY || this.flags['force-tty'];
+
     await this.promptForFromEmail();
-    const validFromEmail = this.validateEmail(this.fromEmail);
     await this.promptForToEmail();
-    const validToEmail = this.validateEmail(this.toEmail);
     await this.promptForSubject();
     await this.promptForText();
-    const sendInformation = { to: validToEmail, from: validFromEmail[0], subject: this.subjectLine, text: this.emailText, html: '<p>' + this.emailText + '</p>' };
-    const attachmentVerdict = await this.askAttachment();
-    await this.promptAttachment(attachmentVerdict);
-    if (this.attachment) {
-      const fileContent = this.readFile(this.attachment);
-      const attachment = this.createAttachmentArray(fileContent);
+
+    if (!this.isTTY && (!this.flags.to || !this.flags.text || !this.subjectLine || !this.fromEmail)) {
+      this.logger.error('All flags must be provided to send email.');
```

Comment:
Should this message make clear that the flags are required only when !tty?

Thinking about a scenario where I run a test script locally that works, then fails when I push to CI.